src/main/java/com/ust/sourcecourse/configuration/service/DBDataSourceService.java

```diff
@@ -150,12 +156,19 @@ public List<DBTableColumn> searchColumnsByTag(String tag) {
 			sourceColumns = sourceColumnRepository.retrieveByTag(tag.toLowerCase());
 
 		}
+		if (sourceColumns.isEmpty()) {
+			throw new ResourceNotFoundException("No tables found with the given tag.");
+		}
 		return sourceColumns.stream().map(SourceColumn -> getDBTableColumn1(SourceColumn)).collect(Collectors.toList());
+
 	}
 
 	public List<String> addTagSourceTable(Long uid, List<String> tags, String description) {
 		SourceTable sourceTable = sourceTableRepository.findById(uid)
 				.orElseThrow(() -> new ResourceNotFoundException("sourcetable", "id", uid));
+		if (tags == null) {
```

Comment:
No need to check this alone. Check if tags and description are empty. Then throw exception. If any one is not empty, then it is valid.

Reply:
resolved